lib/rubystats/uniform_distribution.rb

```diff
@@ -0,0 +1,70 @@
+require 'rubystats/probability_distribution'
+# This class provides an object for encapsulating uniform distributions
+module Rubystats
+  class UniformDistribution < Rubystats::ProbabilityDistribution
+    include Rubystats::SpecialMath
+
+    # Constructs a uniform distribution (defaults to zero lower and
+    # unity upper bound).
+    def initialize(lower=0.0, upper=1.0)
+      lower,uppper = upper,lower if lower > upper
+      @lower = lower.to_f
+      @upper = upper.to_f
+      @pdf_denominator = 1.0 / (@upper - @lower)
+      @use_last = nil
+    end
+
+    # Returns the mean of the distribution
+    def get_mean 
+      return 0.5*(@lower + @upper)
+    end
+
+    # Returns the standard deviation of the distribution
+    def get_standard_deviation
+      return Math.sqrt(get_variance)
+    end
+
+    # Returns the variance of the distribution
+    def get_variance
+      return 1.0/12.0 * (@upper-@lower)**2
+    end
+
+    private
+
+    # Obtain single PDF value
+    # Returns the probability that a stochastic variable x has the value X,
+    # i.e. P(x=X)
+    def get_pdf(x)
+      if x >= @lower and x <= @upper then
```

Comment:
Typically [`&&` is used in ruby instead of `and`](https://github.com/bbatsov/ruby-style-guide#no-and-or-or) and no need for the `then`. Here and below on line 49.